Marlin/Marlin_main.cpp

```diff
@@ -1007,6 +1007,7 @@ inline void get_serial_commands() {
         if (M110) {
           char* n2pos = strchr(command + 4, 'N');
           if (n2pos) npos = n2pos;
+          else { gcode_LastN = strtol(npos + 1, NULL, 10); return;}
```

Comment:
Close. But this skips the checksum test. I think we just need to iron out the logic. If there is an `M110` it should set the current line number. And the existing code does that much, just below. Then, the line number comparison is skipped. And finally, after the checksum test succeeds, `gcode_LastN` is set to `gcode_N`. The next line to come through without `M110` in it will have to have the next number in sequence.

If the `M110` command line isn't getting down to `gcode_LastN = gcode_N` then it implies the checksum on the line is bad.